### PR TITLE
Fix: Add back-to-top button

### LINKED ISSUE #2425: Add Back To Top button
### Have you checked that your feature request isn't already filed?

- [X] I read through [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my feature request.
- [X] Yes, I have checked that this feature request isn't already filed.

### Description & Motivation

Add back to top button for post.

Some example:
- https://jeffreytse.github.io/jekyll-theme-yat/

image

### Pitch

when user browse the post content to any position except the top of the article, `back_to_top` button will appear in the lower left corner of the screen. If user touch the button, it will go back to the top of article.

### Alternatives

_No response_

### Additional context

I will contribute for this feature as possible as I can.

### PR DESCRIPTION
**Pull request by abhilesh**
Fixes #2425 

PR #2427 adds a back-to-top button, however the button overlaps with the footer when `footer_fixed: false` and [has inadequate spacing](https://github.com/alshedivat/al-folio/issues/2425#issuecomment-2121670658) with `footer_fixed: true` 

Changes in this PR:
- Fix positioning of button on fixed and sticky footer layouts 
- Add option to disable back-to-top button by setting `back_to_top: false` in `_config.yml`
- Add button transparency to avoid button blocking content view
- Reduce size of button

Demo - 

| Device | Fixed footer | Sticky footer |
| :-----------: | :-------------: | :-----------: |
|  Mobile | fixed_footer_mobile |  sticky_footer_mobile |
| Desktop | fixed_footer_desktop | sticky_footer_desktop |

Miscellaneous change - Added personal website under `Academics` to `README.md`

### CODE REVIEW

File: _sass/_themes.scss [hunk 1]
```diff
@@ -79,7 +79,7 @@ html[data-theme="dark"] {
   --global-distill-app-color: #{$grey-color-light};
   --global-divider-color: #424246;
   --global-card-bg-color: #{$grey-900};
-  --global-back-to-top-bg-color: #{$white-color};
+  --global-back-to-top-bg-color: rgba(#{red($white-color)}, #{green($white-color)}, #{blue($white-color)}, 0.5);
```
Comment (george-gca) [inline _sass/_themes.scss:82]: Same here `rgba($white-color, 0.5);`

**Review by george-gca — CHANGES_REQUESTED** [review]

**Comment by simonwei97** [comment]
`back-to-top` souce js file is [here](https://github.com/alshedivat/al-folio/blob/master/assets/js/vanilla-back-to-top.min.js), which target repo is https://github.com/vfeskov/vanilla-back-to-top.

**Comment by george-gca** [comment]
Yes, but which changes were made to that file in your commit?

File: _sass/_themes.scss [hunk 2]
```diff
@@ -17,7 +17,7 @@
   --global-divider-color: rgba(0, 0, 0, 0.1);
   --global-card-bg-color: #{$white-color};
   --global-highlight-color: #{$red-color-dark};
-  --global-back-to-top-bg-color: #{$black-color};
+  --global-back-to-top-bg-color: rgba(#{red($black-color)}, #{green($black-color)}, #{blue($black-color)}, 0.4);
```
[the remark this answers is not in the document]
Comment (abhilesh) [reply _sass/_themes.scss:20]: SASS variables don't seem directly compatible with the rgba() function, hence to apply transparency I have to extract the RGB components. 

The suggestion `rgba($black-color, 0.4)` results in this - 
Screenshot 2024-05-24 at 07 02 36

File: _sass/_themes.scss (hunk 1 above)
Comment (abhilesh) [reply _sass/_themes.scss:82]: Same here - 

Screenshot 2024-05-24 at 07 02 58

**Comment by simonwei97** [comment]
> `back-to-top` souce js file #2425 is [here](https://github.com/alshedivat/al-folio/blob/master/assets/js/vanilla-back-to-top.min.js), which target repo is https://github.com/vfeskov/vanilla-back-to-top.

@abhilesh  Like what georage said in this code review, could you explain what do you change for this `back-to-top` js file? Adding transparency config?

File: assets/js/vanilla-back-to-top.min.js
[the remark this answers is not in the document]
Comment (abhilesh) [reply assets/js/vanilla-back-to-top.min.js:1]: I added the following modifications to the already minified `vanilla-back-to-top.min.js` added to the repo in PR #2427 

Changed the dimensions of the button -
`o=Math.round(.43*E),t=Math.round(.29*E)` to `o=Math.round(.35*E),t=Math.round(.2*E)`

Added the following to ensure that the button stays on top of sticky or fixed footer - 

```js
window.addEventListener('load', function() {
    var button = document.getElementById('back-to-top');
    var stickyFooter = document.querySelector('.sticky-bottom');
    var fixedFooter = document.querySelector('.fixed-bottom');
  
    if (!button) {
      return;
    }
  
    window.addEventListener('scroll', function() {
      if (stickyFooter) {
        var stickyFooterRect = stickyFooter.getBoundingClientRect();
        if (window.scrollY + window.innerHeight > document.body.scrollHeight - stickyFooterRect.height) {
          button.style.position = 'fixed';
          button.style.bottom = (stickyFooterRect.height + 20) + 'px'; // 20 is the offset from the footer
        } else {
          button.style.position = 'fixed';
          button.style.bottom = '20px'; // original position
        }
      }
      
      if (fixedFooter) {
        var fixedFooterRect = fixedFooter.getBoundingClientRect();
        button.style.position = 'fixed';
        button.style.bottom = (fixedFooterRect.height + 20) + 'px'; // 20 is the offset from the footer
      }
    });
});
```

Briefly, 
- Get the button and footer elements upon website 'load'
- Upon 'scroll', get the position and size of the footer
- Check if footer visible on screen, if visible, fix the button's position to be 20px above footer.

I, then, minified the script again, just to be consistent with the previous PR #2427

**Review by abhilesh — COMMENTED** [review]
The changes suggested for specifying the colour variables for the button didn't work as expected (see attached screenshots) 

I elaborated on the changes made to the js file.
The original file committed in the previous PR was already minified (and hence a one-liner). I added the code specified, minified the file and committed to the PR. If needed, I can beautify the file and add it to the PR

**Review by george-gca — APPROVED** [review]

**Pull request merged**